### DISCUSSION TITLE
[@types/react-instantsearch-dom] DynamicWidgets - update types

### DISCUSSION THREAD

**Discussion by emil-datcu** [DefinitelyTyped/DefinitelyTyped#57783] [Issues with a @types package]
Hi,
Can someone update the types for react-instantsearch-dom?
The library is now at 6.18, but the types are at just 6.12 and the DynamicWidgets component for example cannot be used because the types don't have it:
![image](https://user-images.githubusercontent.com/90839342/146762371-d5c6a0b6-b054-4c19-8829-74e582b46ac8.png)

[a comment by an automated account is not shown]

**Comment by gburgett** [comment]
Hi Emil!

Would you consider adding that type definition to the types?  These type declarations are maintained by the community and not directly owned by the Algolia team.  We need help to keep them up to date!

Contributing is very simple, and you can contribute just this once without having to add yourself as a "maintainer" of these types.

**Reply by emil-datcu** [reply]
Ahm, I guess I can try, but I won't promise anything, never did this before honestly :P is there a guideline or something that I can look into for starting this?
Just clone the repo, do the changes and submit a PR?

**Reply by emil-datcu** [reply]
Hi Gordon, well, I tried, but I have honestly no idea how/from where to start fixing this and/or how to actually test it... I tried to add just the DynamicWidgets (directly inside my project), but it doesn't seem to do much except removing the error, but not actually working, maybe I'm missing something?

**Reply by gburgett** [reply]
Hi Emil, thanks for trying!  Even if you just started by creating a definition for it like `export class DynamicWidgets extends React.Component<any>`, that's helpful.

To get it into the DefinitelyTyped repository, you will need to fork DefinitelyTyped to your own Github account, then create a pull request with that change.  There's more information here: https://github.com/DefinitelyTyped/DefinitelyTyped#edit-an-existing-package

**Comment by martin-badin** [comment]
I created PR for that but the package has its own types. It will be better to stop contributing to this repository and start contributing to the original repository.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57943
